biometricks/src/main/java/com/handstandsam/biometricks/Biometricks.kt

```diff
@@ -160,5 +163,19 @@ sealed class Biometricks {
                 }
             }
         }
+
+        /**
+         * Checks if we can securely authenticate, i.e. we have secure biometrics hardware and the user is
+         * enrolled. [androidx.biometric.BiometricManager.canAuthenticate] is unusable for this for a couple
+         * of reasons:
+         * 1. On api 28 it falls back to [FingerprintManager.hasEnrolledFingerprints] as the system method
+         * does not exist. However, there may still be non-fingerprint biometrics on those devices so it'll
+         * return a false-negative.
+         * 2. On api 29+ it checks for any form of biometrics, not just ones that are 'secure', so we can
+         * get a false-positive if the user is enrolled in an insecure form of biometrics.
+         */
+        fun canSecurelyAuthenticate(context: Context): Boolean {
```

Comment:
Agreed, @handstandsam have any thoughts about how this should be structured?

Reply:
Also, unfortunately, I don't think we can return an enum instead of a boolean as I don't know of a good way to differentiate between having no secure hardware and having secure hardware with no biometrics enrolled.

Reply:
I am in love with sealed classes, but enums are fine for some cases like this potentially.  However when you talk about the exception that @yogurtearl said we should allow to be thrown, I would want a sealed class result with `UnhandledException(e: Exception)` so that I could be aware this could happen.  In kotlin, discovering that exceptions doesn't work well.

Reply:
Regarding API Design.  Trying to match `BiometricManager` that mirrors the normal API is something to definitely consider.  We'd have to see if it accomplishes everything we need, but in general that would allow clients to onboard easier.

Then again, if we allow them to switch, they may expect equivalent behavior, which we may or may not want to provide.  Creating our own API allows us to have more free range.

Depends on the goals of this (which I very sporadically have time to address, sorry for the delay here!)